.github/workflows/ci.yml

```diff
@@ -268,6 +268,9 @@ jobs:
   publish-docker-master:
     needs: [debian-build, build, integration-tests]
     if: github.event_name == 'push' && github.ref == 'refs/heads/master'
+    strategy:
+      matrix:
+        storage-provider: ["", "-s3", "-azure", "-gcs"]
```

Comment:
Personal preference, but I would've set "s3", "azure" here and later on to the Dockerfile the ``Dockerfile-${{``

Reply:
That was my go to as well, but it wouldn't work for the Uber image which has no prefix. Didn't want to get into conditionals here and keep a simple matrix for all images.